src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlConnectionX.cs

```diff
@@ -22,19 +25,58 @@ namespace Microsoft.Data.SqlClientX
     [DesignerCategory("")]
     internal sealed class SqlConnectionX : DbConnection, ICloneable
     {
-        //TODO: reference to internal connection
+        private SqlCredential? _credential;
+        private SqlDataSource? _dataSource;
+        private SqlInternalConnectionX? _connection;
+        private string _connectionString = string.Empty;
```

Comment:
Very tentative, but is there any point keeping a separate variable here? `_dataSource.ConnectionString` would seem to cover this, and perhaps we could plumb the properties like `ConnectionTimeout` through to it (so that we only go through the parse process once.)

Reply:
I think it's an interesting point. We need to expose the setter so that the connection string can be changed, but that could be used immediately to resolve to a new data source and avoid storing the string locally. For now, let me add a TODO and we can try it out along with the changes that will allow SqlConnectionX to create/resolve its own data source.